include/fi_util.h

```diff
@@ -651,4 +655,24 @@ int ofi_shm_map(struct util_shm *shm, const char *name, size_t size,
 		int readonly, void **mapped);
 int ofi_shm_unmap(struct util_shm *shm);
 
+/*
+ * Name Server TODO: add support for Windows OS
+ * (osd/windows/pthread.h should be extended)
+ */
+
+struct util_ns {
+	RbtHandle	ns_map;
+	pthread_t	ns_thread;
+	int		ns_port;
+};
+
+void ofi_util_ns_start_server(struct util_ns *ns);
+void ofi_util_ns_stop_server(struct util_ns *ns);
+int ofi_util_ns_add_local_name(struct util_ns *ns, int service,
+			       void *name, size_t name_len);
+int ofi_util_ns_del_local_name(struct util_ns *ns, int service,
+			       void *name, size_t name_len);
+void *ofi_util_ns_resolve_name(struct util_ns *ns, const char *server,
+			       int *service);
```

Comment:
Sound reasonable, but in this case we should sent separate msg with `service` data as it is doing for `name`. But some extension - maybe it would be better to use `void *` instead `char *` for `service`?